.golangci.yml

```diff
@@ -1,5 +1,5 @@
 run:
-  deadline: 2m
+  timeout: 10m
```

Comment:
10 minutes is a huge bump, could we perhaps squeeze it down to 5 minutes and the jobs would still pass?

Reply:
Wow, I just noticed that the timeout in the configuration file is being overwritten by the --timeout flag in the Makefile:
https://github.com/kubernetes-sigs/metrics-server/blob/0dcee07ea614e8354ec740dfc3f528a572cacec4/Makefile#L244

So, this is not a bump at all :) :) :) 

I just removed the `--timeout` flag from Makefile, it makes it easier to manage the golangci-lint configurations from one place.

Reply:
I tried to squeeze it down to 5 minutes, but the jobs failed due to exceeding the timeout:
echo from https://prow.k8s.io/view/gs/kubernetes-ci-logs/pr-logs/pull/kubernetes-sigs_metrics-server/1633/pull-metrics-server-verify/1903279512716578816

> golangci/golangci-lint info found version: 1.64.8 for v1.64.8/linux/amd64
/home/prow/go/bin/golangci-lint run || (echo 'Run "make update"' && exit 1)
level=error msg="Timeout exceeded: try increasing it by passing --timeout option"

Reply:
Oh I see, I wasn't aware that we were overriding it in the Makefile, good catch!